zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateParallelGatewayTest.java

```diff
@@ -0,0 +1,108 @@
+    final String sourceProcessId = helper.getBpmnProcessId();
+    final String targetProcessId = helper.getBpmnProcessId() + "_target";
+
+    final String nonExistingElJobTypeExp = "=source_execution_listener_job_type";
+    final String existingElJobTypeExp = "=target_execution_listener_job_type";
+    final String existingElJobTypeVar = "target_execution_listener_job_type";
+    final String existingElJobTypeValue = "elJobType";
+
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(sourceProcessId)
+                    .startEvent()
+                    .parallelGateway("parallel1")
+                    .zeebeStartExecutionListener(nonExistingElJobTypeExp)
+                    .endEvent("end1")
+                    .moveToLastGateway()
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .parallelGateway("parallel2")
+                    .zeebeStartExecutionListener(existingElJobTypeExp)
+                    .endEvent("end2")
+                    .moveToLastGateway()
+                    .endEvent()
+                    .done())
+            .deploy();
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(sourceProcessId)
+            .withVariable(existingElJobTypeVar, existingElJobTypeValue)
+            .create();
+
+    final var incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId("parallel1")
+            .getFirst();
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("parallel1", "parallel2")
+        .migrate();
+
+    RecordingExporter.incidentRecords(IncidentIntent.MIGRATED)
+        .withRecordKey(incident.getKey())
+        .await();
+
+    // then
+    ENGINE.incident().ofInstance(processInstanceKey).withKey(incident.getKey()).resolve();
+    ENGINE.job().ofInstance(processInstanceKey).withType(existingElJobTypeValue).complete();
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementType(BpmnElementType.END_EVENT)
+                .limit(2))
+        .extracting(r -> r.getValue().getElementId())
+        .describedAs(
+            "Expected to successfully evaluate execution listener job type and resolve incident")
+        .contains("end2");
```

Comment:
❓ I don't understand the last assertion. Why only check that one of the two end events was reached? Why not both?

Reply:
I attached other end event to the gateway only for using the parallel gateway in its purpose (running flows in parallel). I expect validating that we are able to migrate the incident on the gateway and being able to move forward in the flow after resolving the incident should be sufficient to test the intended behaviour.